third_party/nvidia/lib/Dialect/NVWS/Transforms/InsertTmemAref.cpp

```diff
@@ -585,10 +646,27 @@ LogicalResult insertTmemAref(TmemAccessDag &accessDag) {
             !getDisallowAccMultiBuffer(wsLoop);
         isMultiStaged = isMultiStaged ? *isMultiStaged && accIsMultiBuffered
                                       : accIsMultiBuffered;
+        if (*isMultiStaged) {
+          // if isMultiStaged: verify we have enough TMEM blocks for it
+          auto hasEnoughTmem = haveEnoughTmem(mmaOp, numTmemBlocks);
```

Comment:
Fold this into the `accIsMultiBuffered` definition (Ideally this logic should belong to `isAccMultibufferingPossible` but SWP...)